runtime/near-vm-logic/src/logic.rs

```diff
@@ -2191,7 +2191,7 @@ impl<'a> VMLogic<'a> {
         let buf = self.get_vec_from_memory_or_register(ptr, len)?;
         self.gas_counter.pay_base(utf8_decoding_base)?;
         self.gas_counter.pay_per(utf8_decoding_byte, buf.len() as u64)?;
-        let account_id = AccountId::from_utf8(buf).map_err(|_| HostError::BadUTF8)?;
+        let account_id = AccountId::try_from(buf.as_slice()).map_err(|_| HostError::BadUTF8)?;
```

Comment:
looks like this changes the protocol

Reply:
> change in serialization format (not sure whether we guarantee that AccountId serializes in the same way as before)

It's a rather simple struct, so it's represented in the same way. And, so far, from the tests I've executed, both serde + borsh serialization formats remain consistent.

> Reject some input that was not previously rejected.

Before working on this PR, I verified that all `AccountId`-s currently used on both `mainnet` and `testnet` were all valid to ensure a case like this doesn't happen. Although, I realize there could be offline cases like these, but I think, if it's gonna be used, it shouldn't have been invalid anyways. The tests had some cases like these, but they've all been updated to use valid AccountIds.

> Return an error where we previously do not return any error

Returning an error from ID validation already happens, just further down the line e.g https://github.com/near/nearcore/blob/2ca2684ad7cf400b5c617f62ac118aa105f16193/runtime/runtime/src/verifier.rs#L40-L42
This change would allow us ensure validation preemptively even before interacting with nearcore. In some cases, it could help reduce the effect of returning Errors wherever applicable. Effectively, we shift the responsibility of validation over to the user instead of propagating that through like we previously did. And in cases like jsonrpc, we can return more quantitative errors regarding invalid Ids instead of situations where it feels like your query got lost in the wind unbeknownst to you that there was an issue with ID validation. cc @frol